Userland/Libraries/LibCore/File.h

```diff
@@ -68,6 +68,12 @@ class File final : public SeekableStream {
     virtual ErrorOr<size_t> seek(i64 offset, SeekMode) override;
     virtual ErrorOr<void> truncate(size_t length) override;
 
+    bool is_directory() const;
+    bool is_device() const;
+    bool is_block_device() const;
+    bool is_char_device() const;
+    bool is_link() const;
+
```

Comment:
I'm not sure how I feel about having those functions here, they would presumably be much more universal if they were in `Core::FileSystem` and taking an `fd` argument. `Core::File` already has an `fd` function that returns the currently open file descriptor, so those can be easily combined without needing to put them into the `Core::File` class itself.